data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/configuration/OnErrorOption.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.configuration;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public enum OnErrorOption {
+    DELETE_MESSAGES("delete_messages"),
+    RETAIN_MESSAGES("retain_messages");
+
+    private static final Map<String, OnErrorOption> OPTIONS_MAP = Arrays.stream(OnErrorOption.values())
+            .collect(Collectors.toMap(
+                    value -> value.option,
+                    value -> value
+            ));
+
+    private final String option;
+
+    OnErrorOption(final String option) {
+        this.option = option.toLowerCase();
+    }
+
+    @JsonCreator
+    static OnErrorOption fromOptionValue(final String option) {
+        return OPTIONS_MAP.get(option.toLowerCase());
```

Comment:
I recommend that you do not lowercase enums. This allows for invalid options to be set:

```
on_error: ReTaIn
```

I find it better to fail if the user enters something like that.